src/ceph-disk

```diff
@@ -2191,11 +2353,65 @@ def mount_activate(
 
     path = mount(dev=dev, fstype=fstype, options=mount_options)
 
+    # check if the disk is deactive, change the journal owner, group
+    # mode for correct user and group.
+    if os.path.exists(os.path.join(path, 'deactive')):
+        # logging to syslog will help us easy to know udev triggered failure
+        if not reactivate:
+            unmount(path)
+            syslog.syslog(syslog.LOG_ERR, 'OSD deactivated! reactivate with: --reactivate')
+            raise Error('OSD deactivated! reactivate with: --reactivate')
+        # flag to activate a deactive osd.
+        deactive = True
+    else:
+        deactive = False
+
```

Comment:
the lines below should not be necessary. If they are, there is a problem elsewhere. It is possible for activation to fail because of a permission problem, but it's a transient failure and activation will be tried again. The chown is done by the journal udev event and the data activation is guaranteed to be tried after the journal udev event changed owner.